presto-parser/src/test/java/com/facebook/presto/sql/parser/TestSqlParser.java

```diff
@@ -1113,6 +1113,19 @@ public void testRenameSchema()
     }
 
     @Test
+    public void testUnicodeString()
+            throws Exception
+    {
+        assertExpression("U&'hello\\6d4B\\8Bd5\\+10FFFFworld\\7F16\\7801'", new StringLiteral("hello\u6d4B\u8Bd5\uDBFF\uDFFFworld\u7F16\u7801"));
+        assertExpression("U&'\u6d4B\u8Bd5ABC\\6d4B\\8Bd5'", new StringLiteral("\u6d4B\u8Bd5ABC\u6d4B\u8Bd5"));
```

Comment:
Add tests for `UESCAPE` as well (`!` is a good choice), along with tests for an invalid escape character such as a plus sign or hex character (see specification).